alg/teca_lapse_rate.py

```diff
@@ -0,0 +1,168 @@
+import numpy as np
+
+def calculate_lapse_rate(t,z,zs,zmax = 9000):
+    """ Calculates the mean lapse rate.
+    
+        input:
+        ------
+        
+            t  : atmospheric temperature [K]
+            
+            z  : geopotential height [m]
+            
+            zs : surface geopotential height [m]
+            
+            zmax : the maximum height to consider in the lapse rate calculation [m]
+            
+        output:
+        -------
+        
+            dtdz : the mean lapse rate [K/m]
+    
+    
+    Uses 1st order finite difference to estimate the lapse rate at all available levels and then averages. Levels
+    below the surface are masked, and levels above zmax are masked.
+    
+    It is assumed that the vertical dimension is the first dimension (e.g., [level, lat, lon]) for both t and z.
+    
+    """
+    
+    # check shapes
+    assert t.shape == z.shape, "t and z must have the same shape; shape(t) = {t.shape} but shape(z) = {z.shape}"
+    assert z[0,...].shape == zs.shape, "All but the first dimension of z must match the dimensions of zs"
```

Comment:
It is OK to leave the assert if this is only for debugging, but if this is a condition that is likely to occur in practice then we probably should always test and raise an exception.

Reply:
this one didn't get addressed yet, see newer comment on that piece of code